src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -277,11 +286,22 @@ protected SnapshotResult<YugabyteDBOffsetContext> doExecute(ChangeEventSourceCon
       Set<String> snapshotCompletedPreviously = new HashSet<>();
 
       for (Pair<String, String> entry : tableToTabletIds) {
-        schemaNeeded.put(entry.getValue(), Boolean.TRUE);
-
-        previousOffset.initSourceInfo(entry.getValue(), this.connectorConfig);
-        this.tabletToExplicitCheckpoint.put(entry.getValue(), null);
-        LOGGER.debug("Previous offset for tablet {} is {}", entry.getValue(), previousOffset.toString());
+        // We can use tableIdToTable.get(entry.getKey()).isColocated() to get actual status.
+        YBPartition p = new YBPartition(entry.getKey() /* tableId */,
+                                        entry.getValue() /* tabletId */, true /* colocated */);
+
+        GetCheckpointResponse resp = this.syncClient.getCheckpoint(
+          tableIdToTable.get(entry.getKey()), this.connectorConfig.streamId(), entry.getValue());
```

Comment:
Colocated snapshot change in addition to the feature supporting resuming of snapshot if stopped midway.